### ISSUE TITLE
Don't want to provide email to you or app

### ISSUE THREAD

**Issue by Gunni** [cla-assistant/cla-assistant#730]
URL to the linked Repo/Org: Not relevant

Steps to reproduce the problem:
1. Open any CLA
2. Click sign
3. Observe you need to provide all github emails to the app/whoever receives the notification that I signed

What is the expected result?
I do not want to provide the emails.

What happens instead?
It demands email for no reason.

Just link my account ID and be done with it.

**Comment by KharitonOff** [comment]
Generally CLA assistant doesn't request user email in order to collect the agreement. As you expect account ID is sufficient. Nevertheless we give other companies and projects, that are using our service, an option to collect further user information in accordance to their need/legal regulations, which might include your email address. Since you haven't provided a link to the corresponding PR we can hardly say whether this is also the case there. 
Besides this, I would like to point you to the fact that with every commit on github you share your email publicly anyways, e.g. here https://api.github.com/repos/advania/pass/commits/36d5064c9da928c8aa943c8df705066b33ca0c21

**Comment by Gunni** [comment]
What you find when you happen to scrape the internet is _different_ than what a user provides to you directly, providing it directly is sometimes interpreted as a consent to getting spammed. Not saying that you do that but i try to avoid it where possible.

What I dislike the most is that the auth page says it's gonna send all emails on my account, that may include emails i do not want to share.

Merge request https://github.com/inverse-inc/packetfence/pull/6461

**Comment by KharitonOff** [comment]
Oh, now I see from your PR that you are talking about the scope of the app token... Yes, we request access to the (minimal possible) user information in order to be able to identify the signee as committer. 

![Screen Shot 2021-08-30 at 16 52 53](https://user-images.githubusercontent.com/8480657/131358901-c99521f8-02eb-409b-8d50-553cea57ce95.png)

Off course it is your right to reject it and not to provide this information if you don't trust the application.

**Comment by Gunni** [comment]
Providing you with the email address on the commit is NO PROBLEM.

Providing you with ALL EMAILS ON MY ACCOUNT is the problem.

**Comment by KharitonOff** [comment]
@Gunni yes, I understand your point and it is absolutely valid! I personally try to avoid sharing my email addresses as well...
Unfortunately we can't do much here. We rely on the token authentication solution from GitHub and there we request from signees the least amount of access to their profile. Here you can see what kind of information an application can request https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps

**Comment by Gunni** [comment]
Yeah why not use

> (no scope) | Grants read-only access to public information (including user profile info, repository info, and gists)

And then authenticate that user? Or if you really want to use email, prompt for an email address?

**Comment by KharitonOff** [comment]
Thanks for your proposal. The team will for sure think about it and see whether it is feasible to reduce the scope to (no scope).

**Comment by Shegox** [comment]
While the current logic doesn't allow to request the scope after the fact, you can manually do that.
On the OAuth-authorization screen you can just remove the `&scope=user%3Aemail` from the url and hit enter. Now only public scopes are required and we don't see your emails. 

A more proper implementation (maybe later) would construct the OAuth URL based on what the CLA requires. If they require the email, add the emails scope if not then omit it.

https://github.com/cla-assistant/cla-assistant/blob/28660d8e2a14c58b04c604878668a16430eb7a35/src/client/controller/cla.js#L22-L47

**Comment by Gunni** [comment]
> On the OAuth-authorization screen you can just remove the `&scope=user%3Aemail` from the url and hit enter. Now only public scopes are required and we don't see your emails.

That was very helpful, thank you. Got the CLA signed. That solves my problem.

**Comment by jsejcksn** [comment]
> Thanks for your proposal. The team will for sure think about it and see whether it is feasible to reduce the scope to (no scope).

Hi @KharitonOff: Do you have any information to report at this time about progress on this? Thanks!

**Comment by Shegox** [comment]
@jsejcksn in the meantime I have taken over cla-assistant from @KharitonOff, but so far I haven't had time to look into it. Sorry. However if you're interested in implementing it, I'm more than happy to help you out/review that!

**Comment by Gunni** [comment]
> [@jsejcksn](https://github.com/jsejcksn) in the meantime I have taken over cla-assistant from [@KharitonOff](https://github.com/KharitonOff), but so far I haven't had time to look into it.

4 years later ... 🦗

**Comment by ShadowRZ** [comment]
@Shegox Please respond after 4 years
